### DISCUSSION TITLE
Setting up Edge Function Webhook without exposing service role key or hardcoding url

### DISCUSSION THREAD

**Discussion by ChuckJonas** [supabase/supabase#14997] [Questions]
If you setup a webhook to an edge function, there is an option to add a "service role key" as the authorization header:

<img width="757" alt="demo___supabase_callawaycloudconsulting_com_s_Org___Supabase" src="https://github.com/supabase/supabase/assets/5217568/fae0ecae-c45b-417a-89d7-1946e52e0061">

I was hoping this would inject the key at runtime, but it just seems to be a pure convenience feature and just results in the same output as if you had pasted it in yourself.

**There are a couple problems:**
1. This creates an additional exposure risk on the service role key.  Anytime I run `npx db diff --linked` I now see the key in plaintext.  This makes it super likely to get checked in to source control!
2. if you have to rotate your key, it will not be updated
3. This doesn't really work with in a multi-enviroment setup (this is also an issue with how it hard codes the function URL)

At the very least, it seems like it should take advantage of the "vault" functionality and store/retrieve an encrypted copy of the key.  Ideally there would be someway to pull it from the ENV

**Comment by j4w8n** [comment]
RE: 1. I thought adding the key in the supabase ui would be fine. How is the key making it's way to your code?

**Reply by ChuckJonas** [reply]
we use migrations and developers may make changes in the supabase app, then use `supabase db diff` to pull down those changes...  If you have added a webhook an authorization header, it will show up in the diff output.
